restapi/src/main/java/io/stargate/web/docsapi/resources/DocumentResourceV2.java

```diff
@@ -660,12 +660,9 @@ public Response getDocPath(
             logger.debug(json);
             return Response.ok(json).build();
           } else {
+            int pageSize = Math.max(1, pageSizeParam);
             final Paginator paginator =
-                new Paginator(
-                    dbFactory.getDataStore(),
-                    pageStateParam,
-                    pageSizeParam,
-                    pageSizeParam > 0 ? pageSizeParam : docsApiConfiguration.getSearchPageSize());
+                new Paginator(dbFactory.getDataStore(), pageStateParam, pageSize, pageSize);
```

Comment:
Why does this one get `pageSize` twice? Just curious.

Would it make sense to have another constructor that takes 3 args?